inspirehep/records/marshmallow/base.py

```diff
@@ -11,43 +11,62 @@
 from marshmallow.schema import Schema
 
 
-class InspireBaseSchema(RecordSchemaJSONV1):
-    # FIXME: this should be renamed to ``uuid`` depends on ``inspire-next``
-    id_ = fields.String(attribute="pid.object_uuid")
+class InspireEnvelopeSchema(RecordSchemaJSONV1):
+    """This schema is the envelope of the API.
 
+    Invenio API returns the following structure:
 
-class InspireBaseMetadataSchema(Schema):
-    _post_dumps = []
+    Examples:
+        {
+            id
+            metadata
+            created
+            updated
+        }
 
-    @post_dump(pass_original=True)
-    def process_post_dump_in_order(self, object_, original_data):
-        for dump_func in self._post_dumps:
-            object_ = dump_func(object_, original_data)
-        return strip_empty_values(object_)
+    Note:
+        In our case we need to enchance this with extra data.
+    """
+
+    id_ = fields.String(attribute="pid.object_uuid")
 
 
-class InspireESEnhancementSchema(Schema):
+class InspireElasticSearchBaseSchema(Schema):
+    """ElasticSearch specific extra data."""
+
     _created = fields.DateTime(dump_only=True, attribute="created")
     _updated = fields.DateTime(dump_only=True, attribute="updated")
 
 
-class InspireIncludeAllFieldsSchemaMixin:
-    """Include all fields from a record."""
+class InspireBaseSchema(Schema):
+    """Base class for Inspire Schema.
+
+    By default it will include all the fields. If you want fields to be excluded
+    you should specify ``exclude``.
+    """
+
+    post_dumps = []
+    """Here we registered all the post_dumps, in order to run them in order."""
 
     def __init__(self, *args, **kwargs):
         super().__init__()
-        self._post_dumps.append(self.include_original_fields)
+        self.post_dumps.append(self.include_original_fields)
 
     def include_original_fields(self, object_, original_data):
         for key, value in original_data.items():
             if key not in object_ and key not in self.exclude:
                 object_[key] = original_data[key]
         return object_
 
+    @post_dump(pass_original=True)
+    def process_post_dump_in_order(self, object_, original_data):
```

Comment:
why not all `object_`, `data`?

Reply:
This might be confusing, as `object_` here is always a dict and `original_data` might be an original object. Maybe they should be called `data` and `original_object`?